apps/glusterfs/pendingop_entry.go

```diff
@@ -291,6 +291,61 @@ func (p *PendingOperationEntry) RecordRemoveDevice(d *DeviceEntry) {
 	p.Type = OperationRemoveDevice
 }
 
+// RecordChild adds or replaces a child operation for the current
+// pending operation entry. Both child and parent can only have
+// one parent/child relationship. Both are updated.
+func (p *PendingOperationEntry) RecordChild(child *PendingOperationEntry) {
+	// track childs id on parent
+	parentsChild := findChange(p.Actions, OpChildOperation)
+	if parentsChild < 0 {
+		a := PendingOperationAction{Change: OpChildOperation, Id: child.Id}
+		p.Actions = append(p.Actions, a)
+	} else {
+		p.Actions[parentsChild].Id = child.Id
+	}
+	// track parent's id on child
+	childsParent := findChange(child.Actions, OpParentOperation)
+	if childsParent < 0 {
+		a := PendingOperationAction{Change: OpParentOperation, Id: p.Id}
+		child.Actions = append(child.Actions, a)
+	} else {
+		child.Actions[childsParent].Id = p.Id
+	}
+}
```

Comment:
This function does not seem to cover all the cases. In case of a RecordChild where the parent already had a child recorded wouldn't we leave the pointer to the parent intact on the old child?

Reply:
Discussed this verbally. I don't think that level of defensive programming is needed at this time. If a "stray" child has an invalid back link to the parent that's a bug in the calling code, not this function.